Service/src/main/java/org/gusdb/wdk/service/filter/CheckLoginFilter.java

```diff
@@ -47,21 +44,29 @@ public class CheckLoginFilter implements ContainerRequestFilter, ContainerRespon
   @Override
   public void filter(ContainerRequestContext requestContext) throws IOException {
 
-    // skip user check for prometheus metrics endpoint to avoid a guest user being created for every request
-    if (requestContext.getUriInfo().getPath().equals(PROMETHEUS_ENDPOINT_PATH)) return;
+    // skip certain endpoints to avoid a guest user being created for those endpoints
+    String requestPath = requestContext.getUriInfo().getPath();
+    if (isPathToSkip(requestPath)) return;
 
     ApplicationContext context = ContextLookup.getApplicationContext(_servletContext);
     RequestData request = ContextLookup.getRequest(_servletRequest.get(), _grizzlyRequest.get());
 
     Optional<CookieBuilder> newCookie =
         CheckLoginFilterShared.calculateUserActions(
-            findLoginCookie(requestContext.getCookies()), context, request.getSession());
+            findLoginCookie(requestContext.getCookies()),
+            ContextLookup.getWdkModel(context),
```

Comment:
i don't understand why there is a change to looking up the context, rather than using `context`

Reply:
Changed the API to pass in a WdkModel rather than the full ApplicationContext since that is all it needed.